src/arrayMethodJoin.js

```diff
@@ -3,9 +3,28 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this.length - 1 === i) {
+        if (this[i] !== undefined) {
+          str += this[i];
+        }
+      } else if (this[i] === null || this[i] === undefined) {
+        str += separator;
+      } else if (separator === null) {
+        str += this[i] + 'null';
+      } else if (separator === undefined) {
+        str += this[i] + ',';
+      } else {
+        str += this[i] + separator;
+      }
+    }
```

Comment:
try to prettify it, get rid of 'if else' and move str += this[i] out of conditions as it's almost common command